runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -4001,23 +4001,51 @@ class INTERPRETER_CLASS
 		return EXECUTE_BYTECODE;
 	}
 
-	/* ToDo: unimplemented - https://github.com/eclipse-openj9/openj9/issues/13614 */
 	/* jdk.internal.misc.Unsafe: public native <V> V uninitializedDefaultValue(Class<?> clz); */
 	VMINLINE VM_BytecodeAction
 	inlUnsafeUninitializedDefaultValue(REGISTER_ARGS_LIST)
 	{
+		j9object_t cls = *(j9object_t*)_sp;
+
 		updateVMStruct(REGISTER_ARGS);
-		Assert_VM_unreachable();
+		buildInternalNativeStackFrame(REGISTER_ARGS);
+
+		j9object_t result = NULL;
+
+		/* TODO (#14073): update this function to have the same behavior as OpenJDK when cls is null or not a vlauetype (currently OpenJDK segfaults in both those scenarios) */
+		if (NULL != cls) {
+			J9Class *j9clazz = J9VM_J9CLASS_FROM_HEAPCLASS(_currentThread, cls);
+			if (J9_IS_J9CLASS_VALUETYPE(j9clazz)) {
+				result = j9clazz->flattenedClassCache->defaultValue;
+			}
+		}
+
+		restoreInternalNativeStackFrame(REGISTER_ARGS);
+		returnObjectFromINL(REGISTER_ARGS, result, 2);
 		return EXECUTE_BYTECODE;
 	}
 
-	/* ToDo: unimplemented - https://github.com/eclipse-openj9/openj9/issues/13614 */
 	/* jdk.internal.misc.Unsafe: public native <V> long valueHeaderSize(Class<V> clz); */
 	VMINLINE VM_BytecodeAction
 	inlUnsafeValueHeaderSize(REGISTER_ARGS_LIST)
 	{
+		j9object_t cls = *(j9object_t*)_sp;
+
 		updateVMStruct(REGISTER_ARGS);
-		Assert_VM_unreachable();
+		buildInternalNativeStackFrame(REGISTER_ARGS);
+
+		I_64 result = (I_64)0;
+
+		/* TODO (#14073): update this function to have the same behavior as OpenJDK when cls is null or not a vlauetype (currently OpenJDK segfaults in both those scenarios) */
+		if (NULL != cls) {
+			J9Class *j9clazz = J9VM_J9CLASS_FROM_HEAPCLASS(_currentThread, cls);
+			if (J9_IS_J9CLASS_VALUETYPE(j9clazz)) {
+				result = (I_64)J9VMTHREAD_OBJECT_HEADER_SIZE(_currentThread);
```

Comment:
> I think this should also take into account pre-padding J9CLASS_PREPADDING_SIZE so the unsafe field offsets make sense.

I guess the pre-padding size can be included either in the header size or in the field offset.

Reply:
I think as long as one is able to index into a flattened array using offsets properly then its fine. But I think you need to take into account the prepadding to do this.

Basically something like the following:

```
FlattenedVTArray arr = ...

unsafe.putInt(arr, ARRAY_HEADER_OFFSET + (elementIndex * (unsafe.getObjectSize(...) - unsafe.valueHeaderSize())) 

```

Reply:
so `unsafe.getObjectSize(...) - unsafe.valueHeaderSize()` must equal the flattened field size

Reply:
OK. then it seems easier/more reasonable to put J9CLASS_PREPADDING_SIZE in the header size.

Reply:
so to be clear: this just means that valueHeaderSize should return `J9CLASS_PREPADDING_SIZE(...) + J9VMTHREAD_OBJECT_HEADER_SIZE(...)` ?

Reply:
yes. Also, when you write the tests, please ensure you have an example like the one above.

Reply:
hmm, I'm not sure how to write an example like the one above because `getObjectSize` isn't implemented yet. Should I just hardcode it for the test?

Reply:
That example does not need to be added for now. It can be added after `getObjectSize()` is implemented.

You can either use a TODO or an issue to make sure you don't forget this.

Reply:
alright, I've taken J9CLASS_PREPADDING_SIZE into account in 0cc55a9